### PR TITLE
add Palm LLM 

### PR DESCRIPTION
**Pull request by jerryjliu**
# Description

Adds an implementation of Palm, from following their API docs.

Adds a unit test too - mocks palm package.

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

### CODE REVIEW

File: llama_index/llms/palm.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        # get num_output
+        self._num_output = num_output or self._model.output_token_limit
+
+        self._generate_kwargs = generate_kwargs
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        """Get LLM metadata."""
+        # TODO: google palm actually separates input and output token limits
```
Comment (logan-markewich) [inline llama_index/llms/palm.py:47]: should we raise a warning if a user tries to set num_output higher than the limit?

File: llama_index/llms/palm.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        Args:
+            prompt (Prompt): Prompt to use for prediction.
+
+        Returns:
+            str: The predicted answer.
+
+        """
+        raise NotImplementedError(
+            "PaLM does not support streaming completion in LlamaIndex atm."
```
Comment (logan-markewich) [inline llama_index/llms/palm.py:85]: They actually don't support it in general yet from what I could tell -- maybe someday they will answer me on github lol
https://github.com/google/generative-ai-python/issues/43

**Review by logan-markewich — APPROVED** [review]
Nice, thanks for getting this in!

**Comment by jerryjliu** [comment]
might hold on landing until @Disiok cleans up the interface and lands it to main

**Pull request merged**